### DISCUSSION TITLE
Problem installing scikit-fmm

### DISCUSSION THREAD

**Discussion by none-of-my-names-is-available** [usnistgov/fipy#1060] [Q&A]
Hello

Trying to get FiPy operational, I tried to follow the demand to install scikit-fmm.
On pip install scikit-fmm, I got (among many other lines):
-       **Need python for x86, but found x86_64
-       Run-time dependency python found: NO (tried sysconfig)
-       ..\meson.build:45:3: ERROR: Python dependency not found**

So, it appears, I either need a new python or a new scikit-fmm?

Many thanks for any help.

Best regards

Günter

**Comment by wd15** [comment]
As far as I'm aware Scikit-fmm is no longer tested as part of FiPy. I looked through the test environments and it doesn't seem to be included. @guyer can correct me on that. I would be very skeptical of it working with FiPy if it's not part of an automate test environment.

Do you specifically need scikit-fmm or something because it is not a requirement for FiPy?

**Reply by none-of-my-names-is-available** [reply]
My target is to simulate transport of polluted particles or gases in the athmosphere. So, I asked chatpgt to help to write a code  using python. It turned me immediately to FiPy, and it was a matter of less than two hours work to get a first version  of a 3d simulation up and running, where the results looked promising. So, the next step is to implement irregular geometries, like a building. So I asked chatpgt to consider a cube an the bottom side of the cartesian mesh and keep it empty.

This was achieved by deleting the gas in this cube in every time step. This was s.th. like a drain and not what I was looking for; I did not want any gas to enter the cube from the beginning,  but have it flow around it. So, I need an irregular surface at the bottom of my mesh.

After looking through some forums, I found an example in two dimensions, which apparently achieved this using the DistanceVariable object. So I tried to use this object in my demo code. And when I tried to import this latter, FiPy complained, it had no scikit-fmm (or more precisely, a library was misssing which is part of scikit-fmm). And I did not succeed to install it, as I mentioned in my issue.

Am I on the right track?

**Reply by wd15** [reply]
That's super interesting using ChatGPT to use FiPy.

I'm a little confused about why the distance function is necessary from your description. If you can describe some geometry with a function then you can use that function to mask some of the domain with a variable that acts as a mask. Coefficients in your equation can then be dependent of the mask. A distance function is only necessary if the distance from the interface (maybe where the mask changes sign or 0.5 level set depending on how the mask is chosen) is required for some of the calculations (e.g. if a diffuse interface formulation is used). 

If you do require a distance function then maybe use Scikit-fmm as stand alone module. It is relatively easy to use the calculated fields from Scikit-fmm in FiPy as they both use Numpy arrays and the Scikit-fmm grid points can be mapped to the FiPy grid points (or cells).
